unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExperiencesConfirmation/ExperiencesConfirmationPopupController.cs

```diff
@@ -0,0 +1,120 @@
+using DCL.World.PortableExperiences;
+using System;
+using System.Collections.Generic;
+
+namespace DCL.PortableExperiences.Confirmation
+{
+    public class ExperiencesConfirmationPopupController
+    {
+        private readonly IExperiencesConfirmationPopupView view;
+        private readonly DataStore dataStore;
+        private readonly IUserProfileBridge userProfileBridge;
+        private readonly List<string> descriptionBuffer = new ();
+
+        private Action acceptCallback;
+        private Action rejectCallback;
+        private string experienceId;
+        private bool dontShowAnymore;
+
+        public ExperiencesConfirmationPopupController(IExperiencesConfirmationPopupView view,
+            DataStore dataStore,
+            IConfirmedExperiencesRepository confirmedExperiencesRepository,
+            IUserProfileBridge userProfileBridge)
+        {
+            this.view = view;
+            this.dataStore = dataStore;
+            this.userProfileBridge = userProfileBridge;
+
+            view.Hide(true);
+
+            view.OnAccepted += () =>
+            {
+                view.Hide();
+
+                if (dontShowAnymore)
+                    confirmedExperiencesRepository.Set(experienceId, true);
+
+                acceptCallback?.Invoke();
+            };
+
+            view.OnRejected += () =>
+            {
+                view.Hide();
+
+                if (dontShowAnymore)
+                    confirmedExperiencesRepository.Set(experienceId, false);
+
+                rejectCallback?.Invoke();
+            };
+
+            view.OnCancelled += () =>
+            {
+                view.Hide();
+                rejectCallback?.Invoke();
+            };
+
+            view.OnDontShowAnymore += () => dontShowAnymore = true;
+            view.OnKeepShowing += () => dontShowAnymore = false;
+
+            dataStore.world.portableExperiencePendingToConfirm.OnChange += OnConfirmRequested;
+        }
+
+        public void Dispose()
+        {
+            view.Dispose();
+
+            dataStore.world.portableExperiencePendingToConfirm.OnChange -= OnConfirmRequested;
+        }
+
+        private void OnConfirmRequested(ExperiencesConfirmationData current, ExperiencesConfirmationData previous)
+        {
+            string pxId = current.Experience.ExperienceId;
+
+            ExperiencesConfirmationData.ExperienceMetadata metadata = current.Experience;
+
+            experienceId = pxId;
+            acceptCallback = current.OnAcceptCallback;
+            rejectCallback = current.OnRejectCallback;
+
```

Comment:
It seems this could be simplified using an async approach:

OnConfirmationRequest should spawn a task that: shows the UI and wait for either accept or reject (UniTaskCompletionSource should help you a lot). It would allow you to remove those ugly `acceptCallback` and `rejectCallback` variables.

Reply:
Did a refactor from callback management to async task. Not sure if its simpler but follows our standards